packages/utils/src/stacktrace.ts

```diff
@@ -64,28 +71,31 @@ export function stripSentryFramesAndReverse(stack: StackFrame[]): StackFrame[] {
 
   let localStack = stack;
 
-  const firstFrameFunction = localStack[0].function || '';
-  const lastFrameFunction = localStack[localStack.length - 1].function || '';
-
-  // If stack starts with one of our API calls, remove it (starts, meaning it's the top of the stack - aka last call)
-  if (firstFrameFunction.indexOf('captureMessage') !== -1 || firstFrameFunction.indexOf('captureException') !== -1) {
-    localStack = localStack.slice(1);
+  if (stack.length > STACKTRACE_LIMIT) {
+    localStack = stack.slice(0, STACKTRACE_LIMIT);
   }
 
   // If stack ends with one of our internal API calls, remove it (ends, meaning it's the bottom of the stack - aka top-most call)
-  if (lastFrameFunction.indexOf('sentryWrapped') !== -1) {
-    localStack = localStack.slice(0, -1);
+  const lastFrameFunction = localStack[localStack.length - 1].function || '';
+  if (/sentryWrapped/.test(lastFrameFunction)) {
+    localStack.pop();
   }
 
   // The frame where the crash happened, should be the last entry in the array
-  return localStack
-    .slice(0, STACKTRACE_LIMIT)
-    .map(frame => ({
-      ...frame,
-      filename: frame.filename || localStack[0].filename,
-      function: frame.function || '?',
-    }))
-    .reverse();
+  localStack = localStack.reverse();
+
+  // If stack starts with one of our API calls, remove it (starts, meaning it's the top of the stack - aka last call)
+  const firstFrameFunction = localStack[localStack.length - 1].function || '';
+  if (/captureMessage|captureException/.test(firstFrameFunction)) {
+    localStack.pop();
```

Comment:
We trim stack top and root if it is a sentry frame fn - if we perform sort between those two calls then we can use .pop in both cases and avoid allocating the array holding shallow copies.